qiskit/providers/v2/job.py

```diff
@@ -0,0 +1,140 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2017.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Job abstract interface."""
+
+from abc import ABC, abstractmethod
+from typing import Callable, Optional
+import time
+
+from qiskit.providers.jobstatus import JobStatus, JOB_FINAL_STATES
+from qiskit.providers.exceptions import JobTimeoutError
+from qiskit.providers.v2.backend import Backend
+
+
+class Job:
+    """Base common type for all versioned Job abstract classes.
+
+    Note this class should not be inherited from directly, it is intended
+    to be used for type checking. When implementing a provider you should use
+    the versioned abstract classes as the parent class and not this class
+    directly.
+    """
+    version = 0
+
+
+class JobV1(Job, ABC):
+    """Class to handle jobs
+
+    This first version of the Backend abstract class is written to be mostly
+    backwards compatible with the v1 providers interface. This was done to ease
+    the transition for users and provider maintainers from v1 to v2. Expect,
+    future versions of this abstract class to change the data model and
+    interface.
+    """
+    _async = True
+
```

Comment:
This also needs `version = 1`

Reply:
Done in: https://github.com/Qiskit/qiskit-terra/pull/5086/commits/0faaee159abf409558a40757ed44136cc65bc0bb